evaluator/src/main/kotlin/PackageRule.kt

```diff
@@ -0,0 +1,152 @@
+     */
+    fun hasLicense() =
+        object : RuleMatcher {
+            override val description = "hasLicense()"
+
+            override fun matches() =
+                pkg.concludedLicense?.licenses()?.isNotEmpty() == true
+                        || pkg.declaredLicenses.isNotEmpty()
+                        || detectedLicenses.isNotEmpty()
+        }
+
+    /**
+     * A [RuleMatcher] that checks if the [package][pkg] is [excluded][Excludes].
+     */
+    fun isExcluded() =
+        object : RuleMatcher {
+            override val description = "isExcluded()"
+
+            override fun matches() = ruleSet.ortResult.isExcluded(pkg.id)
+        }
+
+    /**
+     * A [RuleMatcher] that checks if the [identifier][Package.id] of the [package][pkg] belongs to one of the provided
+     * [orgs][Identifier.isFromOrg].
+     */
+    fun isFromOrg(vararg names: String) =
+        object : RuleMatcher {
+            override val description = "isFromOrg(${names.joinToString()})"
+
+            override fun matches() = pkg.id.isFromOrg(*names)
+        }
+
+    /**
+     * A [RuleMatcher] that checks if the [identifier type][Identifier.type] of the [package][pkg] equals [type].
+     */
+    fun isType(type: String) =
+        object : RuleMatcher {
+            override val description = "isType($type)"
+
+            override fun matches() = pkg.id.type == type
+        }
+
+    /**
+     * A DSL function to configure a [LicenseRule] and add it to this rule.
+     */
+    fun licenseRule(name: String, licenseView: LicenseView, block: LicenseRule.() -> Unit) {
+        val licenses = licenseView.licenses(pkg, detectedLicenses.map { it.license })
+
+        licenses.forEach { (license, licenseSource) ->
+            val findings = if (licenseSource == LicenseSource.DETECTED) {
+                ruleSet.ortResult.collectLicenseFindings()[pkg.id].orEmpty()
+            } else {
+                emptyMap()
+            }
+
+            licenseRules += LicenseRule(name, license, licenseSource, findings).apply(block)
+        }
+    }
+
+    /**
+     * A [Rule] to check a single license of the [package][pkg].
+     */
+    inner class LicenseRule(
+        name: String,
+
+        /**
+         * The license to check.
+         */
+        val license: String,
+
+        /**
+         * The source of the license.
+         */
+        val licenseSource: LicenseSource,
+
+        /**
+         * The associated [LicenseFinding]s. Only used if [licenseSource] is [LicenseSource.DETECTED].
+         */
+        val licenseFindings: Map<LicenseFinding, List<PathExclude>> = emptyMap()
+    ) : Rule(ruleSet, name) {
+        override val description = "\tEvaluating license rule '$name' for $licenseSource license '$license'."
+
+        override fun errorSource() = "$name - $license ($licenseSource)"
+
+        /**
+         * A [RuleMatcher] that checks if the [license] is a valid [SpdxLicense].
+         */
+        fun isSpdxLicense() =
```

Comment:
Why do we need that?

Reply:
To check if a license is a valid SPDX license in a license rule.